### PR TITLE
Cache Appraisal dependencies across CI builds

### PR DESCRIPTION
**Pull request by nickcharlton**
Previously, `appraisal install` didn't pass through a `path` argument.
[This is in progress][1]. [This continues the work started
previously.][2]

So far, this duplicates the existing cache restore/set directives to see
if we get the behaviour intended.

[1]: https://github.com/thoughtbot/appraisal/pull/165
[2]: https://github.com/thoughtbot/administrate/pull/1482

### CODE REVIEW

File: gemfiles/rails42.gemfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ end
 
 group :test do
   gem "ammeter"
-  gem "capybara", "3.30.0"
```
Comment (pablobm) [inline gemfiles/rails42.gemfile]: Is the Capybara update relevant?

Comment (nickcharlton) [reply gemfiles/rails42.gemfile]: No. I'll remove it before I merge it. I'm most interested here on whether or not it makes a build difference / actually works

**Comment by nickcharlton** [comment]
This fails on Ruby 2.6 on CI (only) and I cannot work out why.

**Comment by pablobm** [comment]
The build appears to be passing now. Has anything changed?

**Comment by nickcharlton** [comment]
I wonder if we were getting the reasonably regular failure here and I rebased it?

I'm going to do that again, see if it still passes and look back at what I did: it's possible it's just working now because the feature it's intended to do is just disabled.

**Comment by nickcharlton** [comment]
Okay, coming back to this now by undoing it so it goes back to the feature working as it was supposed to.

**Comment by nickcharlton** [comment]
Closing this, to potentially revisit in future.

It'd be great to cache the additional gems too but it already was quite a bit of work to get going. We can revisit it after GitHub Actions is up and running.

**Pull request closed without merging**